src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -341,6 +391,37 @@ internal static bool TryCreate(SyntheticBoundNodeFactory F, MethodSymbol method,
             return true;
         }
 
+        private static MethodSymbol GetCustomCreateMethod(
+            SyntheticBoundNodeFactory F,
+            NamedTypeSymbol builderType)
+        {
+            // The Create method's return type is expected to be builderType.
+            // The WellKnownMembers routines aren't able to enforce that, which is why this method exists.
+            //
+            // TODO: consider removing WellKnownMember.System_Runtime_CompilerServices_AsyncTaskMethodBuilder*__Create
+            // and using GetCustomCreateMethod for the Task types as well as for custom tasklikes.
```

Comment:
Removed in https://github.com/dotnet/roslyn/pull/13405/commits/f2cdba83c83d34ddac4875def41674d352fc5b7e#diff-995d1bf33206eef8c980dbd08b8601b0L401